packages/apps/spaces/app/src/hooks/useContactOrUserDisplayData/index.ts

```diff
@@ -0,0 +1,22 @@
+import { useCallback } from 'react';
+import { Contact, User } from '@graphql/types';
+
+export const useContactOrUserDisplayName = () => {
+  return useCallback((contact?: Partial<Contact | User> | null) => {
+    if (!contact) return 'Unnamed';
+
+    if (contact.__typename === 'Contact' && contact?.name) {
+      return contact.name;
+    }
+
+    const name = `${contact?.firstName} ${contact?.lastName}`;
+
+    if (name.trim()) {
+      return name;
+    }
+
+    return contact?.emails?.[0]?.email
+      ? contact?.emails?.[0]?.email
+      : 'Unnamed';
+  }, []);
+};
```

Comment:
Just a suggestion, I've added during the migration [ts-pattern](https://github.com/gvergnaud/ts-pattern) which helped me write those kind of mappers with ease.

If you think it'll make sense to use it for mappers and parsers then you can have a look at the docs or at LastTouchpoint cell component where I've used it.